lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -999,7 +999,7 @@ void fieldsToJson(Names names, JsonGenerator gen) throws IOException {
     public EnumSchema(Name name, String doc, LockableArrayList<String> symbols, String enumDefault) {
       super(Type.ENUM, name, doc);
       this.symbols = symbols.lock();
-      this.ordinals = new HashMap<>();
+      this.ordinals = new HashMap<>(symbols.size());
```

Comment:
Hello!  If you're looking to avoid resizing/rehashing the internal map _entirely_, this probably isn't exactly quite you're looking for.

For example, for 100 symbols, HashMap will allocate a table of 128 `Node<>` instances, and at the 96th or 97th insertion (`~0.75 * 128`), it will resize/rehash the table to 256.

I agree that this change should guarantee *fewer* resize/rehashes (and at most one), however.

Reply:
@RyanSkraba Thanks for the pointer. I updated the PR to avoid a resize without touching the load factor.